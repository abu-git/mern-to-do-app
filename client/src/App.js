import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Redirect } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';

//import Authentiation Service
import authCentralState from './Components/Authentication';

//import Components
import Home from './Components/Home';
import Login from './Components/Login';
import Dashboard from './Components/Dashboard';
import Register from './Components/Register';
import Create from './Components/Create';
import Edit from './Components/Edit';
import Delete from './Components/Delete';
import TaskStatus from './Components/StatusUpdate';


//Higher Order Component to protect private component
const ProtectedRoute = ({ component: Component, ...rest}) => (
  <Route {...rest} render={(props) =>(
    authCentralState.isAuthenticated === true ?
      <Component {...props} /> : <Redirect to={{ pathname: '/login', state: {from: props.location}}}/>
  )}/>
);

class App extends Component{
  render(){
    return(
      <Router>
        <div className="App">
          <div className="container">
            <Route exact path="/" component={Home} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/register" component={Register}/>

            <ProtectedRoute exact path="/dashboard" component={Dashboard}/>
            <ProtectedRoute path="/add" component={Create}/>
            <ProtectedRoute path="/dashboard/update/:id" component={Edit} />
            <ProtectedRoute path="/dashboard/delete/:id" component={Delete}/>
            <ProtectedRoute path="/dashboard/status/:id" component={TaskStatus}/>
          </div>
        </div>
      </Router>
    );
  }
}



export default App;
