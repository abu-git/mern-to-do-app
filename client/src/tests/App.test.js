import React from 'react';
import App from '../App';
import { create } from 'react-test-renderer';

describe('Jest testing of React Components', ()=>{
	test('Testing App Component', () =>{
		let tree = create(<App />);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});