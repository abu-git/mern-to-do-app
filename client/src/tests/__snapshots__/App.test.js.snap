// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest testing of React Components Testing App Component 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="row mt-5"
    >
      <div
        className="col-md-6 m-auto"
      >
        <div
          className="card card-body text-center"
        >
          <h1>
            Task Management Application
          </h1>
          <p>
            Create an account or login
          </p>
          <a
            className="btn btn-primary btn-block mb-2"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
          <a
            className="btn btn-secondary btn-block"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
